### PR TITLE
feat: propagate AlgorithmsInit_service use for algorithms::GeoSvc

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This propagates #1195 to the other geometry users that already use the algorithms interfaces.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue: roll out proof of concept and initial implementation of #1195 to others)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

File: src/algorithms/fardetectors/MatrixTransferStatic.cc [hunk 1]
```diff
@@ -20,13 +20,8 @@
...
 
-void eicrecon::MatrixTransferStatic::init(const dd4hep::Detector* det,
-                                          const dd4hep::rec::CellIDPositionConverter* id_conv,
-                                          std::shared_ptr<spdlog::logger> &logger) {
+void eicrecon::MatrixTransferStatic::init() {
 
-  m_log       = logger;
-  m_detector  = det;
-  m_converter = id_conv;
   //Calculate inverse of static transfer matrix
```
Comment (wdconinc) [inline src/algorithms/fardetectors/MatrixTransferStatic.cc:26]:
```suggestion
```

File: src/algorithms/fardetectors/MatrixTransferStatic.h [hunk 2]
```diff
@@ -40,15 +39,20 @@ namespace eicrecon {
...
-    /** algorithm logger */
-    std::shared_ptr<spdlog::logger>   m_log;
-    const dd4hep::Detector* m_detector{nullptr};
-    const dd4hep::rec::CellIDPositionConverter* m_converter{nullptr};
+    //----- Define constants here ------
+    double aXinv[2][2] = {{0.0, 0.0},
+                          {0.0, 0.0}};
+    double aYinv[2][2] = {{0.0, 0.0},
+                          {0.0, 0.0}};
+
```
Comment (wdconinc) [inline src/algorithms/fardetectors/MatrixTransferStatic.h:52]:
```suggestion
```

File: src/algorithms/fardetectors/MatrixTransferStatic.h [hunk 3]
```diff
@@ -40,15 +39,20 @@ namespace eicrecon {
                             {"outputParticleCollection"},
                             "Apply matrix method reconstruction to hits."} {}
 
-    void init(const dd4hep::Detector* detector, const dd4hep::rec::CellIDPositionConverter* id_conv, std::shared_ptr<spdlog::logger>& logger);
+    void init() final;
     void process(const Input&, const Output&) const final;
 
   private:
```
Comment (wdconinc) [inline src/algorithms/fardetectors/MatrixTransferStatic.h:46]:
```suggestion
```

**Review by veprbl — APPROVED** [review]
Diff LGTM, capybara report shows no changes

**Pull request merged**